tests/CatalogTests/Registration/RegistrationMakerCatalogItemTests.cs

```diff
@@ -81,5 +82,99 @@ private static string GetContentString(StorageContent content)
                 return reader.ReadToEnd();
             }
         }
+
+        public class CreatePageContent_SetsLicenseUrlAndExpression
+        {
+            private readonly Uri _catalogUri = new Uri("http://example/catalog/mypackage.1.0.0.json");
+            private readonly Uri _registrationBaseAddress = new Uri("http://example/registration/");
+            private readonly Uri _packageContentBaseAddress = new Uri("http://example/content/");
+            private readonly Uri _baseAddress = new Uri("http://example/registration/mypackage/");
+            private Graph _graph;
+
+            public CreatePageContent_SetsLicenseUrlAndExpression()
+            {
+                _graph = new Graph();
+                _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.Id),
+                    _graph.CreateLiteralNode("MyPackage"));
+                _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.Version),
+                    _graph.CreateLiteralNode("01.02.03+ABC"));
+                _graph.Assert(
+                    _graph.CreateUriNode(_catalogUri),
+                    _graph.CreateUriNode(Schema.Predicates.Published),
+                    _graph.CreateLiteralNode("2015-01-01T00:00:00+00:00"));
+            }
+
+            [Theory]
+            [InlineData(false, false, false, "", "", "http://gallery.org")]
+            [InlineData(true, false, false, "https://test.org", "", "http://gallery.org")]
+            [InlineData(true, true, false, "http://gallery.org/packages/MyPackage/1.2.3/license", "MIT", "http://gallery.org")]
+            [InlineData(true, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org")]
+            [InlineData(false, true, false, "http://gallery.org/packages/MyPackage/1.2.3/license", "MIT", "http://gallery.org")]
+            [InlineData(false, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org")]
+            [InlineData(false, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org/")]
+            [InlineData(false, false, true, "http://gallery.org/packages/MyPackage/1.2.3/license", "", "http://gallery.org//")]
+            public void CreatePageContent_SetsLicenseUrlAndExpressionProperly(bool hasLicenseUrl, bool hasLicenseExpression, bool hasLicenseFile,
+                                                                             string expectedLicenseUrlValue, string expectedLicenseExpressionValue, string galleryBaseAddress)
```

Comment:
It would be better to pass `licenseUrl`, `licenseExpression` and `licenseFile` as inputs instead of flags, with `null` indicating the absence of an entry. This way we'd be able to test various values of those properties.